src/Symfony/Component/Console/Exception/CommandNotDefinedException.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Exception;
+
+/**
+ * Represents an incorrect command name typed in the console.
+ */
+class CommandNotDefinedException extends \InvalidArgumentException
```

Comment:
You should also create a marker like for most other components like the one in the routing component `Symfony\Component\Routing\Exception\ExceptionInterface` and then implement it here:

`class CommandNotDefinedException extends \InvalidArgumentException implements ExceptionInterface`

That way, it's possible to catch all exceptions coming from the Console.